cr-examples/onnx/src/main/java/oracle/code/onnx/OnnxRuntime.java

```diff
@@ -49,27 +51,154 @@ public final class OnnxRuntime {
         }
     }
 
-    public static <T, U extends Quotable & Supplier<Tensor<T>>> Tensor<T> execute(U codeLambda) {
+    @FunctionalInterface
+    public interface OnnxFunction<T> extends Supplier<T>, Quotable {
+    }
+
+    public static <T> Tensor<T> execute(MethodHandles.Lookup l, OnnxFunction<Tensor<T>> codeLambda) {
```

Comment:
The expectation seems to be that the input values are all tensors... maybe we can model them as explicit parameters in the `OnnxFunction` interface - e.g. something like:

```
interface OnnxFunction<T> {
     T apply(Tensor... tensors)
}
```

This would open the possibility to use method reference in the example -- but one problem is that this uses a raw type (Tensor).

Another option could be to model both the input and output generically:

```
interface OnnxFunction<T, U> {
     T apply(U u)
}
```

Where both T and U can be either:
* a single Tensor (of any type)
* a list/array of Tensors (again, of any type)